test/integration.js

```diff
@@ -327,4 +327,34 @@ describe('mongoose-autopopulate plugin', function() {
     await band.save();
     assert.ok(!band.populated('lead'));
   });
+  it('supports findOneAndDelete', async function() {
```

Comment:
For test failures, you need to update all the `assert.equal(schemaStub.pre.calls.length, 3);` calls in `test/unit.js` to `assert.equal(schemaStub.pre.calls.length, 5);`